### DISCUSSION TITLE
Power Down mode: very different readings between 1614 and 3224

### DISCUSSION THREAD

**Discussion by FerGT50** [SpenceKonde/megaTinyCore#654] [Q&A] [answered]
I'm measuring the power draw of these new AVR chips: AVR128DB28, ATTiny 1614 and ATTiny 3224.

In particular, I'm very interested in ATTiny1614 and 3224 power draw when in Power Down sleep mode; I'm using Arduino 1.8.19 with the latest megaTinyCore 2.5.10.

Running the very same code on the two chips, which is basically

```
set_sleep_mode(SLEEP_MODE_PWR_DOWN);
sleep_enable();
sleep_cpu();
```
(all pins are configured as INPUT_PULLUP and nothing is connected)
I get:
about 0.23 microA at 3.3V for ATTiny 1614 (0.8 microwatt)
about 131 microA at 3.3V for ATTiny 3224 (430 microwatt) 

That is, 3224 draws **_550x_** the power of the 1614 in the same conditions, running the same code, within the same test circuit!
The datasheet reports the same power consumption for the two in Power Down mode (about 0.10 microA at 25 C).
Isn't it strange? I found nothing in the Errata about this specific test case.

**Comment by SpenceKonde** [comment, marked as answer]
Did you turn off the ADC completely?
Have you made sure to power cycle it after last uploading over UPDI (sometimes that can cause problems for power consumption 
 Remember that the power.h library doesn['t do anything on modern AVRs, 

I haven't worked with the power saving modes at all though  - getting a better understanding of that and how to use them effectively is to write those imprved sleep and power management libraries and fix the attach interrupt bug. But that only comes after I get ATTinyCore 2.0.0 released - they haven't had a release in over a year because stuff one these cores kept coming up.  But these things have to time-share with everything else.

**Reply by FerGT50** [reply]
EDIT: My fault.
Turns out I commented out ADC0.CTRLA = 0 because it does nothing on the 1614, but it makes a HUGE difference with the 3224!
Current draw drops from 131 to 0.39 microA!

~~Yes to your two questions.
The strange thing is: I have the very same test board, I just swap the chips (1614 and 3224 are interchangeable, same pinouts; both are soldered to a SOP-to-DIP adapter), and the very same test code.
Yet the 3224 draws 550x the current in Power Down mode. Quite puzzling.~~
Edit: in Run mode, they draw about the same.

**Comment by SpenceKonde** [comment]
Boom! 

Yeah I was thinking, where are the differences that could be relevant there...And the only thing that seemed like it could plausibly be an energy sink on that scale was something related to the ADC... I mean hell, The 2-series lost more peripherals than they gained! 1614 had 2 ADCs, 2-series has one very nice one. The 2-series shed the type D timer, two comparators, a DAC,  while al it gained was a second USART ad 2 more CCL blocks, and better event channels, but none of thosse have a reputation for burning power, but ADCs definitely do, and there's some errata that implies that lowlat mode (we set that by default) doesn't quite behave as it's supposed to.

**Comment by FerGT50** [comment]
By the way, the higher ADC frequency (vs. the older parts) gave me strong headaches a few days ago... my "nanocurrent meter" (via C discharge) showed 300 nA without load when driven by the AVR DB vs the old 328PU.
Turns out reading the C voltage with the ADC at high frequency depletes 1 microF fast (S&H internal capacitor charges from the "power source" every time)! :D
Did not find the value of the internal S&H C by the way, does not seem to be in the datasheet.

**Comment by SpenceKonde** [comment]
Can you explain what you mean there, specifically what the "nanocurrent meter" is - is that AVR running a proram of that name or an external highlly accurate current meter?

**Reply by FerGT50** [reply]
I used to measure currents in the uA/nA range via a custom setup: an AVR running a program which times the discharge of a known capacitor. Inspired by: http://www.technoblogy.com/show?2S67

Now I mainly use a TinyCurrent device: https://www.n-fuse.co/devices/tinyCurrent-precision-low-Current-Measurement-Shunt-and-Amplifier-Device.html

**Reply by SpenceKonde** [reply]
Sec 33.16 of the datasheet, (Electrical properties chapter, ADC section, "Cin, input capacitance" 7 pf with the PGA, 8pf without it. 

Try analogReadEnh with the PGA set to unity gain under the conditions that threw off the measurements before. I would be incredibly interested to hear the results of that test. 

The ADC on the 2-series has a much less pronounced "memory" than the 0/1-series parts or classic AVRs.; I think the effect you saw is a manifestation of that. What's the threshold voltage that you measure to? I bet if you the 2-series running at a voltage on average twice the voltage of the thing you were measuring, you'd see little if any effect. Above Vdd/2 I would expect it to pull the voltage down and inflate the current reading, while below it, I would expect it to pull the voltage upwards, reducing the apparent current. 

(remember what it was like on classic AVRs, where when switching pins code like this was ubiquitous because your results were otherwise skewed towards the previously read pin's value?

```c
analogRead(A0);
analogRead(A0);
data = analogRead(A0);
```

**Reply by SpenceKonde** [reply]
Also, if that doesn't change things, I would be very interested to hear if setting freerun mode in CTRLE  does. (you should be able to kick off a read with analogReadEnh or analogRead after that, then continually poll ADC.STATUS for the RESREADY bit) to read the values in when it does..

I suspect that freerunning mode - or at least freerunning mode with the PGA enabled - would make a big difference. . In freerunning and burst samples, the PGA is supposedly continually sampling the pin, so it shouldn't put as large a load on the pin since it's not recharging the S&H cap every time.
